### PR TITLE
Add option to test for methods like fit or xit

### LINKED ISSUE #42: Support for `xit`?
Wondering if there is a way to add support for `xit`?
```js
xit("some test", () => {})
```

### PR DESCRIPTION
**Pull request by sualko**
fix #42

### CODE REVIEW

**Comment by sualko** [comment]
@atulpatildbz would be nice if you could test my changes.

**Comment by atulpatildbz** [comment]
@sualko Thanks for taking the time to fix this.

I have tested the change and it seems to be working fine

image

```
npx eslint cypress/e2e/test.cy.js

/Users/atul.patil/code/ui-automation/cypress/e2e/test.cy.js
  3:8  error  it.only not permitted  no-only-tests/no-only-tests
  4:5  error  xit not permitted      no-only-tests/no-only-tests

✖ 2 problems (2 errors, 0 warnings)
```

**Comment by sualko** [comment]
@levibuzolic would be awesome if you could look into this pull request

File: README.md [hunk 1]
```diff
@@ -69,4 +69,6 @@ Option | Type | Description
 ---|---|---
 `block` | `string[]` | Specify the block names that your testing framework uses. Add a `*` to the end of any string to enable prefix matching (ie. `test*` will match `testExample.only`)<br>Defaults to `["describe", "it", "context", "test", "tape", "fixture", "serial", "Feature", "Scenario", "Given", "And", "When", "Then"]`
 `focus` | `string[]` | Specify the focus scope that your testing framework uses.<br>Defaults to `["only"]`
+`functions` | `string[]` | Specify not permitted functions. Good examples are `fit` or `xit`.<br>Defaults to `[]` (disabled)
+`suffixes` | `string[]` | Specify filename suffixes which contain your test cases to reduce the number of false positives. Rules are only applied to files which match a suffix from the list.<br>Defaults to `[]` (disabled)
```
Comment (levibuzolic) [inline README.md:73]: Little bit on the fence about this option, this is something that is already available through ESLint's flat config (or `overrides` in legacy configs). 

Both of these options allow you to specify a separate set of rules that only apply to test files which has the added benefit of avoiding running any of this plugin's code on non-test files.

**Comment by levibuzolic** [comment]
Thanks for the contribution @sualko, sorry I missed this earlier.

I think I'd prefer to merge this without the filename filtering as it's already possible to acheive with ESLint's config.

I'm also starting to realise the options API here isn't scaling very well and will look into putting together a new API that allows consumers to be more explicit with how they target their test runners' specific API. But I wont let that halt improving the existing version.

**Comment by sualko** [comment]
It took some time, but I removed the filename filter @levibuzolic

**Comment by sualko** [comment]
@levibuzolic can you look into this again?

**Review by levibuzolic — APPROVED** [review]

**Comment by sualko** [comment]
@levibuzolic :pray: thanks a lot for your work on this nice plugin

**Comment by levibuzolic** [comment]
@sualko I'm just going to add some test cases for the new functionality and will package a release as soon as I'm back at my desk. Thanks again for your contribution.

**Comment by levibuzolic** [comment]
@sualko sorry for the delay, your contribution has been released in [v3.2.0](https://github.com/levibuzolic/eslint-plugin-no-only-tests/releases/tag/v3.2.0)

**Pull request merged**